src/failed-spec-parser.js

```diff
@@ -2,14 +2,13 @@ export default function (output = '') {
   let match = null
   let CUCUMBERJS_TEST = /^\d+ scenarios?/m
   let failedSpecs = new Set()
-  let PROTRACTOR_SHARDED = /------------------------------------/g
+  let MULTI_TESTS = /------------------------------------/g
   let SPECFILE_REG = /.+Specs:\s(.*\.js)/g
 
-  if (PROTRACTOR_SHARDED.test(output) && SPECFILE_REG.test(output)) {
-    console.log('sharded test found')
-    let testsOutput = output.split('------------------------------------')
-    testsOutput.shift()
-    let RESULT_REG = /\d+\sspec|assertions?,\s(\d+)\sfailures?/g
+  if (MULTI_TESTS.test(output) && SPECFILE_REG.test(output)) {
+    let testsOutput = output.split('------------------------------------').slice(1)
```

Comment:
I'll explain the mention, but both options are viable.

The logging contains something like:
Intro
----
log test 1
----
log test 2
conclusion.

Option A: Use slice.
This way you know that each item in testsOutput is a test, because Intro has been sliced out. If the specfile could not be retrieved in the test, an empty entry is added to failedSpecs.

Option B: check if (specfile && result=== failed)
This way you don't ensure that each item in testsOutput is a test. You just check if you can retrieve the failure and the specfile.

Personally I like option A better, because if a scenario exists where we can't determine the specfile using the current regexp, protractor-flake will re-run all tests. With option B it will just be ignored and skip the test (even if it failed).